### ISSUE TITLE
False positive: `VarCouldBeVal`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4736]
## Expected Behavior
A property that is reasigned shouldn't be flagged even if it is inside an unknown type.

## Observed Behavior
It is flagged.

## Steps to Reproduce
```kotlin
@Test
fun `does not report private variables that are re-assigned inside an unknown type`() {
    val code = """
        private var a = 1

        fun foo() {
            with(UnknownReference) {
                a = 2
            }
        }
    """

    assertThat(subject.lintWithContext(env, code)).isEmpty()
}
```

## Context
In my code `UnkownReference` is a `ViewBinding` that detekt right now can't handle. Probably #4723 will help

## Your Environment
* Version of detekt used: 1.20.0 & main

**Comment by chao2zhang** [comment]
+1 I have the same observation as well with `DataBinding`
